record/record.go

```diff
@@ -198,9 +214,34 @@ func (r *Reader) Next() (io.Reader, error) {
 		return nil, r.err
 	}
 	r.started = true
+	r.inRecovery = false
 	return singleReader{r, r.seq}, nil
 }
 
+// Recover attempts to recover from a corrupted record by continuing at the
+// next block boundary. Recover clears internal error flags. Calling Recover
+// before Next, Read, and in practice, ioutil.ReadAll, raises an error, is
```

Comment:
I'm not sure why Recover needs to return an error. You could make it a no-op if there is no error, or if you've already hit io.EOF.

If you've hit io.ErrUnexpectedEOF, it seems like you could allow the caller to call Recover. All that would happen is that they'd get an io.EOF on Next, meaning that there wasn't any good records after recovery was attempted, which sounds perfectly appropriate to me.

This could be:
// Recover clears any errors read so far, so that calling Next will start
// reading from the next good 32KiB block. If there are no such blocks, Next
// will return io.EOF.
//
// Recover also marks the current reader, the one most recently returned by
// Next, as stale.
func (r *Reader) Recover() {
  etc
}
and note that there is no error returned.